### ISSUE TITLE
Clarification of feature extraction and processing

### ISSUE THREAD

**Issue by CP121** [u1234x1234/kaggle-yelp-restaurant-photo-classification#2]
I would be very grateful if you could clarify your feature extraction workflow as much of the file compress.py is commented out. **Is the following correct?**

1. extract features from the training data using n ImageNet pretrained models

2. normalise each feature matrix separately using sklearn.preprocessing.normalize

2. concatenate all feature matrices horizontally (axis=1) 

3. calculate the column mean 'm' (axis=0) and subtract

4. Apply svd using sklearn.decomposition.TruncatedSVD  with n_components=64 and  algorithm='arpack'

5. repeat for the test data: normalise, concatenate, subtract the mean 'm' (calculated from the training data in step 3), transform using svd (which was fit on the training data in step 4).

Thank you.

**Comment by u1234x1234** [comment]
Sorry for the late reply.
Thanks for your interest.
In general you are right, but I've used 7 different feature sets, described [here](blog.kaggle.com/2016/04/28/yelp-restaurant-photo-classification-winners-interview-1st-place-dmitrii-tsybulevskii/#how-did-you-deal-with-the-multi-instance-aspect-of-this-problem), and for the some of feature sets some steps was skipped.  
4. - 5. is essentially PCA.
Features from the model trained on the full ImageNet gave me main boost in accuracy.
If you any questions please feel free to ask me.
